cpp/src/arrow/ipc/ipc-read-write-test.cc

```diff
@@ -86,6 +87,52 @@ TEST(TestMessage, Equals) {
   ASSERT_FALSE(msg5.Equals(msg1));
 }
 
+TEST(TestMessage, SerializeTo) {
+  const int64_t body_length = 64;
+
+  flatbuffers::FlatBufferBuilder fbb;
+  fbb.Finish(flatbuf::CreateMessage(fbb, internal::kCurrentMetadataVersion,
+                                    flatbuf::MessageHeader_RecordBatch, 0 /* header */,
+                                    body_length));
+
+  std::shared_ptr<Buffer> metadata;
+  ASSERT_OK(internal::WriteFlatbufferBuilder(fbb, &metadata));
+
+  std::string body = "abcdef";
+
+  std::unique_ptr<Message> message;
+  ASSERT_OK(Message::Open(metadata, std::make_shared<Buffer>(body), &message));
+
+  int64_t output_length = 0;
+  int64_t position = 0;
+
+  std::shared_ptr<io::BufferOutputStream> stream;
+
+  {
+    ASSERT_OK(io::BufferOutputStream::Create(&stream));
+    ASSERT_OK(message->SerializeTo(stream.get(), 8, &output_length));
+    ASSERT_OK(stream->Tell(&position));
+    ASSERT_EQ(BitUtil::RoundUp(metadata->size() + 4, 8) + body_length, output_length);
+    ASSERT_EQ(output_length, position);
+  }
+
+  {
+    ASSERT_OK(io::BufferOutputStream::Create(&stream));
+    ASSERT_OK(message->SerializeTo(stream.get(), 64, &output_length));
+    ASSERT_OK(stream->Tell(&position));
+    ASSERT_EQ(BitUtil::RoundUp(metadata->size() + 4, 64) + body_length, output_length);
```

Comment:
Adding a comment here would be nice here.

Reply:
Same as above. I'll add a comment